User-facing Workflow Invocation Grid

jmchilton opened the pull request:
- Refactor generic grid out of admin-centric invocation grid introduced in #8664 
- Add contextual menus for target workflow and history based on WorkflowDropdown added in #8551.
- Add workflow and history vuex stores for caching storedworkflow and history data fetched from the server (based on vuex-cache pattern introduced in #8044
- Update all workflow APIs to allow fetching stored workflows from a stored workflow ID or a workflow ID using the new query parameter (``instance=false`` - override default of ``false`` to ``true`` to fetch by a ``model.Workflow`` ID instead of a ``model.StoredWorkflow`` ID).

I had a request from another developer that the basic version of this grid (5d916e3) without the contextual menus should be backported to 19.09 - I think that is reasonable but I also recognize it was already a big ask to get #8664 in.

<img width="297" alt="Screen Shot 2019-09-24 at 10 19 59 AM" src="https://user-images.githubusercontent.com/216771/65520097-0a041f80-deb5-11e9-80ac-2557b36c252f.png">

<img width="1131" alt="Screen Shot 2019-09-24 at 10 20 44 AM" src="https://user-images.githubusercontent.com/216771/65520105-0d97a680-deb5-11e9-81a4-6626dd028337.png">

<img width="1149" alt="Screen Shot 2019-09-24 at 10 20 52 AM" src="https://user-images.githubusercontent.com/216771/65520116-138d8780-deb5-11e9-983f-4937f11ba680.png">

jmchilton committed:
Create a user grid for workflow invocations.

jmchilton committed:
history store and history drop down in invocation grid

jmchilton commented on client/galaxy/scripts/components/admin/Invocations.vue:
```diff
@@ -3,18 +3,16 @@
         <h2 class="mb-3">
```
File should be moved out of ``components/admin``.

guerler commented on client/galaxy/scripts/components/HistoryDropdown.vue:
```diff
@@ -0,0 +1,46 @@
+        onSwitch: function() {
+            // this switches the history, but doesn't trigger a reload yet...
+            // extract as service from history-model?
+            jQuery.getJSON(`${getAppRoot()}history/set_as_current?id=${this.history.id}`);
```
We should request the data without using jquery.

guerler commented on client/galaxy/scripts/components/User/RecentInvocations.vue:
```diff
@@ -0,0 +1,41 @@
+        return {
+            invocationItems: [],
+            loading: true,
+            busy: true,
```
Wondering if and how we use the `busy` attribute here and if it is equivalent to `loading`.

jmchilton replied:
Yeah - eliminated that - good point!

jmchilton committed:
Add workflow store, workflow options to invocations menu.

jmchilton committed:
Tweaks to invocations grid based on comments.

- Eliminate unneeded busy variable.
- Refactor setting current history to share with history-model, enable in grid - eliminate the use of jquery.
- Move Invocations.vue (the base grid) out of components/admin and into components/Workflow/

jmchilton commented:
Rabased and Committed: Tweaks to invocations grid based on comments.

- Eliminate unneeded busy variable.
- Refactor setting current history to share with history-model, enable in grid - eliminate the use of jquery.
- Move Invocations.vue (the base grid) out of components/admin and into components/Workflow/

mvdbeek commented on client/galaxy/scripts/components/Workflow/Invocations.vue:
```diff
@@ -0,0 +1,138 @@
+                    </div>
+                    <div v-else>
+                        <workflow-dropdown :workflow="getWorkflowByInstanceId(data.item.workflow_id)" />
+                        <!--
```
We can drop this, right ?

jmchilton replied:
Yup, I've cleaned this up in a rebase. Thanks!

mvdbeek commented on client/galaxy/scripts/components/History/HistoryDropdown.vue:
```diff
@@ -0,0 +1,44 @@
+    methods: {
+        onSwitch: function() {
+            // this switches the history, but doesn't trigger a reload yet...
+            // extract as service from history-model?
```
You've done that, right ?

jmchilton replied:
I've cleaned this up in a rebase. Thanks!

mvdbeek commented on client/galaxy/scripts/components/History/HistoryDropdown.vue:
```diff
@@ -0,0 +1,44 @@
+    },
+    methods: {
+        onSwitch: function() {
+            // this switches the history, but doesn't trigger a reload yet...
```
I'm not sure if this is correct and at which level to place this, but `Galaxy.currHistoryPanel.switchToHistory(this.history.id);` seems to work in my testing.

jmchilton replied:
Yeah - I think I'm going to revert the service extraction and de-duplication and just use this line. I'm worried about conflicts with other history panel work on going - this seems to be a very clean solution - thanks!

mvdbeek reviewed (commented):
This is so cool! But click on the row (but not on workfow or history) to expand the invocations feels a bit awkward. Maybe a `+` sign on the left would help ?

jmchilton commented:
The row expanding details was copied over from the jobs list interface in the admin UI - but I agree it isn't super clear. I had assumed the repository grid and details worked the same but it doesn't. I'll take another shot at this.

mvdbeek reviewed (approved):
There's a glitch with the drop down appearing higher up than you'd expect it, but having the overview is very nice.
<img width="298" alt="Screenshot 2019-11-19 at 15 51 23" src="https://user-images.githubusercontent.com/6804901/69156935-7cbffe80-0ae4-11ea-9e90-c6d9a555a4c6.png">

martenson commented:
switching to WIP since it has conflicts and outstanding comments

jmchilton committed:
Fix wandering drop down menu - seems incompatible with 'responsive'.

jmchilton commented:
Seems the drop down menu was not compatible with the table being marked 'responsive'. Fixed and rebased to resolve conflicts.

martenson committed:
switch plus/minus to chevrons for expand/collapse elements

martenson commented:
I pushed a tiny icon change. Rest LGTM. Thanks!

mvdbeek commented:
Adding a screenshot for the release notes:
<img width="885" alt="Screenshot 2020-02-04 at 11 33 00" src="https://user-images.githubusercontent.com/6804901/73737075-345ac900-4742-11ea-9f50-0f0e4c3d9271.png">
